src/rgw/rgw_bucket.cc

```diff
@@ -1212,39 +1212,44 @@ int RGWBucket::check_index(RGWBucketAdminOpState& op_state,
   return 0;
 }
 
-
-int RGWBucket::policy_bl_to_stream(bufferlist& bl, ostream& o)
+template <typename T>
```

Comment:
This seems like a highly generalized function, especially given the template, and not bucket-specific. I wonder if something like this belongs as a static function in rgw_bucket.cc. This might lead to the code being duplicated in other sub-areas of RGW. Might this functionality be useful more widely (e.g., rgw_common.h)?

Reply:
agree, I think I can move to rgw_common and tackle these two here. And later catch the other usages of this pattern in another master pr?

Reply:
Yeah, I don't think you need to fix all usages right now. But putting this in a more general place sets us up for the next step. This is cool; thank you!